### DISCUSSION TITLE
Retrieving ATSPM data via NTCIP?

### DISCUSSION THREAD

**Discussion by Brian-Stem** [udotdevelopment/ATSPM#190] [Ideas]
Are there any plans to implement retrieval of the hires datas using NTCIP 1202v3? I know that we still need to support signals in the field that only support old, proprietary retrieval methods, so, we still need to use the existing vendor-specific adapters for the foreseeable future. However, in the spirit of standards conformance, I think, that the FTP, custom file formats, etc. should be obsoleted and deprecated, especially if signal manufacturer claims conformance to the 1202v3.

**Comment by AndrewMC6** [comment]
I would be interested in this also, as most of our signals are behind port-forwarding and I am a bit reluctant to open up FTP, even if a rule limited the IPs that could access it. The NTCIP ports are already open, with community names changed.

**Comment by ChesireFreeThrow** [comment]
We have been looking into this for the future. We have also been testing SFTP as it is more secure and seems more universal on the controllers we've tested then using FTP.
